src/shogun/base/Parameter.cpp

```diff
@@ -3844,7 +3844,7 @@ bool TParameter::copy(TParameter* target)
 				*(char**)target->m_parameter=SG_MALLOC(char, num_bytes);
 				/* check whether ptype is SGOBJECT, if yes we need to initialize
 				   the memory with NULL for the way copy_ptype handles it */
-				if (m_datatype.m_ptype==PT_SGOBJECT)
+				if (m_datatype.m_ptype==PT_SGOBJECT || m_datatype.m_stype==ST_STRING)
```

Comment:
The downstream function `TParameter::copy_stype()` checks `target_ptr->slen` - so this occasionally has strange effects if not initialized I think...